server.js

```diff
@@ -1,34 +1,116 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from 'express';
+import cors from 'cors';
+import listEndpoints from 'express-list-endpoints';
+
+import booksData from './data/books.json';
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(express.json());
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+});
+
+// query to filter on autor, title and language.
+app.get('/books', (req, res) => {
+  const { author, title, language } = req.query
+
+  if (!author && !title && !language) {
+    res.status(404).send("Mandatory parameters are missing");
+    return;
+  }
+
+  let bookList = booksData;
+
+  if (author) {
+    bookList = bookList
+      .filter((book) => book.authors.toLowerCase().includes(author.toLowerCase()))
+  }
+
+  if (title) {
+    bookList = bookList
+      .filter((book) => book.title.toString().toLowerCase().includes(title.toLowerCase()))
+  }
+
+  if (language) {
+    bookList = bookList
+      .filter((book) => book.language_code.toLowerCase().includes(language.toLowerCase()))
+  }
+
+  if (bookList.length) {
+    res.json(bookList)
+  } else {
+    res.status(404).send("No match");
+  }
+});
+
+// endpoint to get the top 5 books sorted on rating
+app.get('/top5', (req, res) => {
+  const books = booksData.sort((a, b) => b.average_rating - a.average_rating);
```

Comment:
you used sorting a few times. maybe it is a good idea to create a sorting function?